app/code/Magento/Theme/Controller/Result/JsFooterPlugin.php

```diff
@@ -39,26 +39,38 @@ public function __construct(ScopeConfigInterface $scopeConfig)
      */
     public function beforeSendResponse(Http $subject)
     {
-        $content = $subject->getContent();
-        $script = [];
-        if (is_string($content) && strpos($content, '</body') !== false) {
-            if ($this->scopeConfig->isSetFlag(
-                self::XML_PATH_DEV_MOVE_JS_TO_BOTTOM,
-                ScopeInterface::SCOPE_STORE
-            )
-            ) {
-                $pattern = '#<script[^>]*+(?<!text/x-magento-template.)>.*?</script>#is';
-                $content = preg_replace_callback(
-                    $pattern,
-                    function ($matchPart) use (&$script) {
-                        $script[] = $matchPart[0];
-                        return '';
-                    },
-                    $content
-                );
-                $subject->setContent(
-                    str_replace('</body', implode("\n", $script) . "\n</body", $content)
-                );
+        $content = $subject->getContent() ?? '';
+
+        $bodyClose = '</body';
+
+        if (is_string($content) && strpos($content, $bodyClose) !== false && $this->scopeConfig->isSetFlag(
+            self::XML_PATH_DEV_MOVE_JS_TO_BOTTOM,
+            ScopeInterface::SCOPE_STORE
+        )) {
+            $scripts = '';
+
+            $scriptOpen = '<script';
+            $scriptClose = '</script>';
+
+            $scriptOpenPos = strpos($content, $scriptOpen);
+            while ($scriptOpenPos !== false) {
```

Comment:
The line breaks here are inconsistent. I think we should remove all line breaks inbetween var declarations and add one before the `while` statement.